### PR TITLE
Refactoring-Magento-AdminNotification-module

### PR DESCRIPTION
**Pull request by agorbulin**
### Description (*)
enrich/fix PHPDocs, small refactoring in Magento_AdminNotification

### Fixed Issues (if relevant)
none

### Manual testing scenarios (*)
not necessary

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @agorbulin. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/AdminNotification/Model/Feed.php [hunk 1]
```diff
@@ -198,7 +198,7 @@ public function setLastUpdate()
     /**
      * Retrieve feed data as XML element
      *
-     * @return \SimpleXMLElement
+     * @return bool|\SimpleXMLElement
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/Feed.php:201]: Please inverse order of types as normally this method returns xml.

File: app/code/Magento/AdminNotification/Model/Inbox.php [hunk 2]
```diff
@@ -7,25 +7,24 @@
...
+ * @method setDescription(string $value)
  * @method string getUrl()
- * @method \Magento\AdminNotification\Model\Inbox setUrl(string $value)
+ * @method setUrl(string $value)
  * @method int getIsRead()
- * @method \Magento\AdminNotification\Model\Inbox setIsRead(int $value)
+ * @method setIsRead(int $value)
  * @method int getIsRemove()
- * @method \Magento\AdminNotification\Model\Inbox setIsRemove(int $value)
+ * @method setIsRemove(int $value)
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/Inbox.php:27]: There should be `$this` return type for all the setters I believe.

File: app/code/Magento/AdminNotification/Model/Inbox.php [hunk 3]
```diff
@@ -41,7 +40,9 @@ protected function _construct()
     }
 
     /**
-     * {@inheritdoc}
+     * @param null $severity
+     *
+     * @return array|mixed|null|string
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/Inbox.php:45]: You can do better :)

File: app/code/Magento/AdminNotification/Model/Inbox.php [hunk 4]
```diff
@@ -134,7 +135,9 @@ public function add($severity, $title, $description, $url = '', $isInternal = tr
      * @param string|string[] $description
      * @param string $url
      * @param bool $isInternal
-     * @return $this
+     *
+     * @return $this|\Magento\Framework\Notification\NotifierInterface
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/Inbox.php:139]: Just `$this`, please. Fix all.

File: app/code/Magento/AdminNotification/Model/ResourceModel/Grid/Collection.php [hunk 5]
```diff
@@ -20,7 +20,7 @@ class Collection extends \Magento\AdminNotification\Model\ResourceModel\Inbox\Co
     /**
      * Add remove filter
      *
-     * @return Collection|\Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection
+     * @return $this|\Magento\AdminNotification\Model\ResourceModel\Inbox\Collection
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/ResourceModel/Grid/Collection.php:23]: Just `$this`.

File: app/code/Magento/AdminNotification/Model/ResourceModel/Inbox/Collection/Critical.php [hunk 6]
```diff
@@ -27,7 +27,9 @@ protected function _construct()
     }
 
     /**
-     * @return $this
+     * Init collection select
+     *
+     * @return $this|\Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/ResourceModel/Inbox/Collection/Critical.php:32]: Just `$this`.

File: app/code/Magento/AdminNotification/Model/ResourceModel/System/Message/Collection.php [hunk 7]
```diff
@@ -26,13 +26,15 @@ class Collection extends \Magento\Framework\Model\ResourceModel\Db\Collection\Ab
...
+     *
      * @param \Magento\Framework\Data\Collection\EntityFactory $entityFactory
      * @param \Psr\Log\LoggerInterface $logger
      * @param \Magento\Framework\Data\Collection\Db\FetchStrategyInterface $fetchStrategy
      * @param \Magento\Framework\Event\ManagerInterface $eventManager
      * @param \Magento\Framework\Notification\MessageList $messageList
-     * @param mixed $connection
-     * @param \Magento\Framework\Model\ResourceModel\Db\AbstractDb $resource
+     * @param \Magento\Framework\DB\Adapter\AdapterInterface|null $connection
+     * @param \Magento\Framework\Model\ResourceModel\Db\AbstractDb|null $resource
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Model/ResourceModel/System/Message/Collection.php:37]: Nope, `null` must not be passed.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@agorbulin , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**